packages/api/amplify_api_dart/test/amplify_api_config_test.dart

```diff
@@ -6,71 +6,72 @@ import 'package:amplify_core/amplify_core.dart';
-    const region = 'us-east-1';
-    const authorizationType = APIAuthorizationType.apiKey;
-    const apiKey = 'abc-123';
-
-    setUpAll(() async {
-      const config = AWSApiConfig(
-        endpointType: endpointType,
-        endpoint: endpoint,
-        region: region,
-        authorizationType: authorizationType,
-        apiKey: apiKey,
-      );
-
-      endpointConfig = const EndpointConfig('GraphQL', config);
+  group('EndpointConfig.getUri', () {
+    test('should return existing uri', () async {
+      const endpoint = 'https://abc123.amazonaws.com/graphql';
+      final endpointConfig = createEndpointConfig(endpoint);
+      final uri = endpointConfig.getUri();
+      expect(uri.toString(), equals(endpoint));
     });
 
-    test('should return valid URI with null params', () async {
+    test('should return existing uri with a trailing slash', () async {
+      const endpoint = 'https://abc123.amazonaws.com/graphql/';
+      final endpointConfig = createEndpointConfig(endpoint);
       final uri = endpointConfig.getUri();
-      final expected = Uri.parse('$endpoint/');
-
-      expect(uri, equals(expected));
+      expect(uri.toString(), equals(endpoint));
     });
-  });
-
-  group('REST Config', () {
-    const endpointType = EndpointType.rest;
-    const endpoint = 'https://abc123.appsync-api.us-east-1.amazonaws.com/test';
-    const region = 'us-east-1';
-    const authorizationType = APIAuthorizationType.iam;
-
-    setUpAll(() async {
-      const config = AWSApiConfig(
-        endpointType: endpointType,
-        endpoint: endpoint,
-        region: region,
-        authorizationType: authorizationType,
-      );
 
-      endpointConfig = const EndpointConfig('REST', config);
+    test('should return existing uri with no path segments', () async {
+      const endpoint = 'https://abc123.amazonaws.com';
+      final endpointConfig = createEndpointConfig(endpoint);
+      final uri = endpointConfig.getUri();
+      expect(uri.toString(), equals(endpoint));
     });
 
-    test('should return valid URI with params', () async {
+    test('should return valid URI with path and params', () async {
+      const endpoint = 'https://abc123.amazonaws.com/test';
       const path = 'path/to/nowhere';
       final params = {'foo': 'bar', 'bar': 'baz'};
+      final endpointConfig = createEndpointConfig(
+        endpoint,
+        type: EndpointType.rest,
+      );
       final uri = endpointConfig.getUri(path: path, queryParameters: params);
-
-      final expected = Uri.parse('$endpoint/$path?foo=bar&bar=baz');
-
-      expect(uri, equals(expected));
+      const expected = '$endpoint/$path?foo=bar&bar=baz';
+      expect(uri.toString(), equals(expected));
     });
 
-    test('should handle a leading slash', () async {
+    test('should handle a path with a leading slash', () async {
+      const endpoint = 'https://abc123.amazonaws.com/test';
       const path = '/path/to/nowhere';
```

Comment:
Can this be `/path/to/nowhere/` to test the behavior when `path` has a trailing slash?

Reply:
Yeah. This is an interesting case. Similar to my comment above, I think the trailing slash should be kept if the consumer explicitly included it. The full uri becomes `https://abc123.amazonaws.com/test/path/to/nowhere/?foo=bar&bar=baz` (with a `/` immediately before the `?`). Is that your expectation?

Reply:
That's what I would expect, yeah

Reply:
Updated